crates/bitwarden-ssh/src/import.rs

```diff
@@ -0,0 +1,279 @@
+                    pkcs8::DecodePrivateKey::from_pkcs8_encrypted_pem(&encoded_key, password)
+                        .map_err(|err| match err {
+                            ed25519::pkcs8::Error::EncryptedPrivateKey(_) => {
+                                SshKeyImportError::WrongPassword
+                            }
+                            _ => SshKeyImportError::ParsingError,
+                        })?
+                }
+                None => ed25519::pkcs8::DecodePrivateKey::from_pkcs8_pem(&encoded_key)
+                    .map_err(|_| SshKeyImportError::ParsingError)?,
+            };
+            let private_key = ssh_key::private::PrivateKey::from(Ed25519Keypair::from(
+                &private_key.secret_key.into(),
+            ));
+            let private_key_openssh = private_key
+                .to_openssh(LineEnding::LF)
+                .map_err(|_| SshKeyImportError::ParsingError)?;
+            Ok(SshKey {
+                private_key: private_key_openssh.to_string(),
+                public_key: private_key.public_key().to_string(),
+                key_fingerprint: private_key.fingerprint(HashAlg::Sha256).to_string(),
+            })
+        }
+        KeyType::Rsa => {
+            let private_key: rsa::RsaPrivateKey = match password {
+                Some(password) => {
+                    pkcs8::DecodePrivateKey::from_pkcs8_encrypted_pem(&encoded_key, password)
+                        .map_err(|err| match err {
+                            pkcs8::Error::EncryptedPrivateKey(_) => {
+                                SshKeyImportError::WrongPassword
+                            }
+                            _ => SshKeyImportError::ParsingError,
+                        })?
+                }
+                None => pkcs8::DecodePrivateKey::from_pkcs8_pem(&encoded_key)
+                    .map_err(|_| SshKeyImportError::ParsingError)?,
+            };
+
+            let private_key = ssh_key::private::PrivateKey::from(
+                RsaKeypair::try_from(private_key).map_err(|_| SshKeyImportError::ParsingError)?,
+            );
+            let private_key_openssh = private_key
+                .to_openssh(LineEnding::LF)
+                .map_err(|_| SshKeyImportError::ParsingError)?;
+            Ok(SshKey {
+                private_key: private_key_openssh.to_string(),
+                public_key: private_key.public_key().to_string(),
+                key_fingerprint: private_key.fingerprint(HashAlg::Sha256).to_string(),
+            })
+        }
+        _ => Err(SshKeyImportError::UnsupportedKeyType),
+    }
+}
+
+fn import_openssh_key(encoded_key: String, password: String) -> Result<SshKey, SshKeyImportError> {
+    let private_key = match ssh_key::private::PrivateKey::from_openssh(&encoded_key) {
+        Ok(k) => k,
+        Err(err) => match err {
+            ssh_key::Error::AlgorithmUnknown
+            | ssh_key::Error::AlgorithmUnsupported { algorithm: _ } => {
```

Comment:
**issue:** If arguments are unused we prefer to use `..` since it's less fragile if the enum pair changes.
```suggestion
            | ssh_key::Error::AlgorithmUnsupported { .. } => {
```